### ISSUE TITLE
[BUG]: IAFD Agent; and, All Agents not returning Actors Headshots

### ISSUE THREAD

**Issue by fourstix** [CodyBerenson/PGMA-Modernized#320]
# Mandatory #
Sample file in log is named (Catalina) - Hot Details and is matching with the GEVI agent.

## Describe the bug: ## 

Actor headshots are no longer appearing.  The searches in IAFD return response error code 403 (Forbidden).

## Index Site URL(s) Attempting to Match: ##
https://gayeroticvideoindex.com/video/18371

## Log Attached: ##
[com.plexapp.agents.GEVI.log](https://github.com/user-attachments/files/17740371/com.plexapp.agents.GEVI.log)

# Optional #

## Screenshot(s) with all nudity redacted: ##
<img width="694" alt="Screenshot 2024-11-13 181421" src="https://github.com/user-attachments/assets/7668f8b4-4717-4782-b640-99227cea579e">

## Additional Context: ##

Log contains several messages like this for each actor:

2024-11-13 18:07:26,679 (7f520063eb38) :  INFO (sandbox:19) - GEVI  -       ----------------------------------------------------------------------------------------------------
2024-11-13 18:07:26,680 (7f520063eb38) :  INFO (sandbox:19) - GEVI  - UTILS :: Unmatched Cast Name            8 - Scott Mann/scottmann
2024-11-13 18:07:26,681 (7f520063eb38) :  INFO (sandbox:19) - GEVI  - UTILS :: Gay/Bisexual/Straight Cast xPath //table[@id="tblFem" or @id="tblMal"]/tbody/tr
2024-11-13 18:07:26,682 (7f520063eb38) :  INFO (sandbox:19) - GEVI  - UTILS :: Request URL                   https://www.iafd.com/ramesearch.asp?searchtype=comprehensive&searchstring=Scott+Mann
2024-11-13 18:07:26,693 (7f520063eb38) :  DEBUG (networking:144) - Requesting 'https://www.iafd.com/ramesearch.asp?searchtype=comprehensive&searchstring=Scott+Mann'
2024-11-13 18:07:27,208 (7f520063eb38) :  ERROR (networking:197) - Error opening URL 'https://www.iafd.com/ramesearch.asp?searchtype=comprehensive&searchstring=Scott+Mann'
2024-11-13 18:07:27,209 (7f520063eb38) :  INFO (sandbox:19) - GEVI  - UTILS :: Request Failed                Try CFScrape
2024-11-13 18:07:27,760 (7f520063eb38) :  ERROR (utils:5484) - GEVI - UTILS :: ❌ File: utils.py, Line: 5714, Function: updateMetadata -> getCast -> matchCast
2024-11-13 18:07:27,760 (7f520063eb38) :  ERROR (utils:5485) - GEVI - UTILS ::    Error:: Cannot Process IAFD Cast Search Results, Scott Mann: 'NoneType' object has no attribute 'xpath'

## Desktop (please complete the following information): ##
 - OS: [e.g. iOS]    QNAP  QTS 5.2.1 (Based on Ubunto Linux)
 - Browser [e.g. chrome, safari]  Firefox
 - Version [e.g. 22]  132.0.1 (64 bit Windows)

**Comment by CodyBerenson** [comment]
@jph71 This is not the cloudflare issue.   Your CFScrape fix continues to bypass the cloudflare 403 problem.  IAFD has changed XPath.  Thanks.

**Comment by fourstix** [comment]
Ah, good catch!  I edited the bug description and removed the parts of the description where I blamed Cloudflare.

**Comment by JPH71** [comment]
That makes it much easier to fix...
I will try and get it sorted by the weekend xxx

**Comment by dimmitri9** [comment]
@JPH71 did you find what's causing the error?

**Comment by JPH71** [comment]
It's a 403 error...
I am looking at using selenium to sort it out...

**Comment by dimmitri9** [comment]
It's still showing up with "?" and grey circles instead headshots

**Comment by JPH71** [comment]
Because it hasn't been fixed... its a cloudflare issue

**Comment by dimmitri9** [comment]
Can I help with something?

**Comment by JPH71** [comment]
Yes, if you can play with the headers OR better still implement Selenium as
that initiates a headless web browser...

Cheers man

**Comment by dimmitri9** [comment]
Is there any workaround out yet @JPH71 ?
